src/shaderlib/lighting/lighting.glsl.js

```diff
@@ -29,10 +29,9 @@ uniform float ambientRatio;
 uniform float diffuseRatio;
 uniform float specularRatio;
 
-float getLightWeight(vec4 position_worldspace, vec3 normals_worldspace) {
+float getLightWeight(vec3 position_worldspace_vec3, vec3 normals_worldspace) {
   float lightWeight = 0.0;
 
-  vec3 position_worldspace_vec3 = position_worldspace.xyz / position_worldspace.w;
```

Comment:
For some reason, this line generates all zeros on Intel Iris Pro GPU used by MacBook Pro (but not other Intel integrated GPU). So I changed the interface of getLightWeight() and moved this division out of this function. 

Normally, the 4th component is 1.0...so this division is not doing much anyway.